.github/actions/wait-for-netlify/index.js

```diff
@@ -0,0 +1,137 @@
+const core = require('@actions/core');
+const axios = require('axios');
+
+const READY_STATES = ['ready', 'current'];
+
+function getNetlifyUrl(url) {
+    return axios.get(url, {
+        headers: {
+            Authorization: `Bearer ${process.env.NETLIFY_TOKEN}`
+        }
+    });
+}
+
+const waitForDeployCreation = (url, commitSha, MAX_TIMEOUT) => {
+    const increment = 15;
+
+    return new Promise((resolve, reject) => {
+        let elapsedTimeSeconds = 0;
+
+        const handle = setInterval(async () => {
+            elapsedTimeSeconds += increment;
+
+            if (elapsedTimeSeconds >= MAX_TIMEOUT) {
+                clearInterval(handle);
+                return reject(`Timeout reached: Deployment was not created within ${MAX_TIMEOUT} seconds.`);
+            }
+
+            const { data: netlifyDeployments } = await getNetlifyUrl(url);
+
+            if (!netlifyDeployments) {
+                return reject(`Failed to get deployments for site`);
+            }
+
+            const commitDeployment = netlifyDeployments.find((d) => d.commit_ref === commitSha);
+
+            if (commitDeployment) {
+                clearInterval(handle);
+                return resolve(commitDeployment);
+            }
+
+            console.log(`Not yet created, waiting ${increment} more seconds...`);
+        }, increment * 1000);
+    });
+};
+
+const waitForReadiness = (url, MAX_TIMEOUT) => {
+    const increment = 30;
+
+    return new Promise((resolve, reject) => {
+        let elapsedTimeSeconds = 0;
+        let state;
+
+        const handle = setInterval(async () => {
+            elapsedTimeSeconds += increment;
+
+            if (elapsedTimeSeconds >= MAX_TIMEOUT) {
+                clearInterval(handle);
+                return reject(
+                    `Timeout reached: Deployment was not ready within ${MAX_TIMEOUT} seconds. Last known deployment state: ${state}.`
+                );
+            }
+
+            const { data: deploy } = await getNetlifyUrl(url);
+
+            state = deploy.state;
+
+            if (READY_STATES.includes(state)) {
+                clearInterval(handle);
+                return resolve();
+            }
+
+            console.log(`Not yet ready, waiting ${increment} more seconds...`);
+        }, increment * 1000);
+    });
+};
+
+const waitForUrl = async (url, MAX_TIMEOUT) => {
```

Comment:
We have library wait-on in our dependencies, you can use that instead

Reply:
I copied `probablyup/wait-for-netlify-action` action and added the commit-sha input parameter as we need it, we probably can rewrite the whole thing in a better manner, but do we want to invest time in it? :)